### PR TITLE
Allow nerfed blazes, endermen to take water damage (fixes #2841)

### LINKED ISSUE #2841: Nerfed Blazes don't take damage in water.
### What behaviour is expected:
Blaze should take damage as they do in vanilla.

### What behaviour is observed:
Blaze doesn't take damage.

### Steps/models to reproduce:
1. Fresh paper 1.15.1 server with no configs changed except nerfing spawned mobs in spigot.yml
2. Place down a blaze spawner and spawn a blaze (I did it in the overworld so I could place water)
3. Submerge the blaze in water.

https://media.discordapp.net/attachments/289587909051416579/665681540872667156/unknown.png

### Plugin list:
None

### Paper build number:
"This server is running Paper version git-Paper-43 (MC: 1.15.1) (Implementing API version 1.15.1-R0.1-SNAPSHOT)"

### Anything else:
This may potentially be in other versions (I only tested it for the above build number but am reporting this for someone who tried some other versions).

### PR DESCRIPTION
**Pull request by Trigary**
In case you believe that the mentioned issue should be addressed here, in Paper, this is my take at doing that.

### CODE REVIEW

**Comment by kickash32** [comment]
You only changed the nerfed mobs patch, correct?

**Comment by Trigary** [comment]
That's right. Well, 17 other files also got unintentionally changed thanks to the rebase.

**Comment by kickash32** [comment]
Next time I would recommend you don't commit the unnecessary changes.

**Comment by Trigary** [comment]
Will do. I thought that these hashes changed for a reason and that stuff would break if they weren't updated, but it would seem that I was wrong.

**Comment by electronicboy** [comment]
This would probably be much more suited towards #2763 (or, an extension thereof)

**Review by kickash32 — CHANGES_REQUESTED** [review]
I think it being in this patch is good because the activation range patch, PR already deals with mobs in water however, if they are nerfed, it wouldn't really change the behaviour if they were activated or otherwise.

I would recommend forcing a diff on the damage entity line, just in case in the future, there are changes to the behaviour of certain mobs touching water. Also I won't what would be the affects of doing mobTick() for nerfed mobs.

**Comment by electronicboy** [comment]
Ignore my comment, my brain is trashed, for some reason I thought this was an EAR thing, not a nerf'd mob thing

**Comment by Trigary** [comment]
> I would recommend forcing a diff on the damage entity line, just in case in the future, there are changes to the behaviour of certain mobs touching water. Also I won't what would be the affects of doing mobTick() for nerfed mobs.

While calling `mobTick()` on nerfed blazes seems to be fine, endermen, on the other hand, attempt random teleports in that method. What do you mean by "forcing a diff on the damage entity line"? Are you suggesting the use of empty, protected methods that subclasses can overwrite instead of `instanceof` checks?

**Comment by kickash32** [comment]
No I mean the lines you copied from the blaze and enderman classes, should be included in the patch. When/if they are changed they cause a conflict, instead of silently changing and causing a bug in the future.

**Comment by Trigary** [comment]
Oh, makes sense. Let me fix this PR.

**Comment by Trigary** [comment]
Okay, so I have undone the unnecessary file changes and created a diff for the enderman class. I didn't create a diff for the blaze class because, to quote electroniccat:
> pulling in a class to add an collision thing is a bit bleh...

**Review by kickash32 — APPROVED** [review]

File: Spigot-Server-Patches/0013-Allow-nerfed-mobs-to-jump-and-take-water-damage.patch [hunk 1]
```diff
@@ -81,5 +109,5 @@ index 0f0dc7277..43908b040 100644
      public void e() {
          if (this.a.getRandom().nextFloat() < 0.8F) {
 -- 
-2.24.0
+2.16.1.windows.4
```
Comment (electronicboy) [inline Spigot-Server-Patches/0013-Allow-nerfed-mobs-to-jump-and-take-water-damage.patch]: 2018 wants der git bak

**Review by electronicboy — APPROVED** [review]

**Review by zachbr — APPROVED** [review]

**Pull request merged**